examples/paho-mqtt/Makefile

```diff
@@ -0,0 +1,41 @@
+APPLICATION = paho-mqtt-example
+
+# If no BOARD is found in the environment, use this default:
+BOARD ?= native
+
+# This has to be the absolute path to the RIOT base directory:
+RIOTBASE ?= $(CURDIR)/../..
+
+# Comment this out to disable code in RIOT that does safety checking
+# which is not needed in a production environment but helps in the
+# development process:
+DEVELHELP ?= 1
+
+# Change this to 0 show compiler invocation lines by default:
+QUIET ?= 1
+
+USEPKG += paho-mqtt
+
+USEMODULE += sock_async_event
+USEMODULE += shell
+USEMODULE += shell_commands
+USEMODULE += ps
+
+# paho-mqtt depends on TCP support, choose the stack you want
+USEMODULE += lwip_ipv6_autoconfig
```

Comment:
The use of IPv4 or IPv6 must be configurable as with other lwip applications, such as in `tests/lwip_sock_tcp`. Otherwise it will not be possible to use IPv4 for netdevs like `esp_wifi` or `stm32_eth`.
```suggestion
LWIP_IPV4 ?= 0

ifneq (0, $(LWIP_IPV4))
  USEMODULE += ipv4_addr
  USEMODULE += lwip_arp
  USEMODULE += lwip_ipv4
  USEMODULE += lwip_dhcp_auto
  CFLAGS += -DETHARP_SUPPORT_STATIC_ENTRIES=1
  LWIP_IPV6 ?= 0
else
  LWIP_IPV6 ?= 1
endif

ifneq (0, $(LWIP_IPV6))
  USEMODULE += ipv6_addr
  USEMODULE += lwip_ipv6_autoconfig
endif

# paho-mqtt depends on lwip for TCP support
```